diff -Naur apache-log4cxx-0.10.0/src/examples/cpp/console.cpp apache-log4cxx-0.10.0.new/src/examples/cpp/console.cpp
--- apache-log4cxx-0.10.0/src/examples/cpp/console.cpp	2008-03-31 22:34:52.000000000 +0000
+++ apache-log4cxx-0.10.0.new/src/examples/cpp/console.cpp	2008-08-03 17:12:55.000000000 +0000
@@ -22,6 +22,7 @@
 #include <log4cxx/logmanager.h>
 #include <iostream>
 #include <locale.h>
+#include <cstring>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
diff -Naur apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp apache-log4cxx-0.10.0.new/src/main/cpp/inputstreamreader.cpp
--- apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp	2008-03-31 22:34:09.000000000 +0000
+++ apache-log4cxx-0.10.0.new/src/main/cpp/inputstreamreader.cpp	2008-08-03 17:13:08.000000000 +0000
@@ -20,6 +20,7 @@
 #include <log4cxx/helpers/exception.h>
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <cstring>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
diff -Naur apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp apache-log4cxx-0.10.0.new/src/main/cpp/socketoutputstream.cpp
--- apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2008-03-31 22:34:09.000000000 +0000
+++ apache-log4cxx-0.10.0.new/src/main/cpp/socketoutputstream.cpp	2008-08-03 16:49:20.000000000 +0000
@@ -19,6 +19,7 @@
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
+#include <cstring>
 
 using namespace log4cxx;
 using namespace log4cxx::helpers;
